composer/callbacks/image_visualizer.py

```diff
@@ -0,0 +1,189 @@
+            ...     optimizers=optimizer,
+            ...     max_duration="1ep",
+            ...     callbacks=[ImageVisualizer()],
+            ... )
+
+    The images are logged by the :class:`.Logger` to the following key(s) as described
+    below.
+
+    +---------------------------------------------+---------------------------------------+
+    | Key                                         | Logged data                           |
+    +=============================================+=======================================+
+    | ``Images/Train``                            |   Sampled examples of train images    |
+    +---------------------------------------------+---------------------------------------+
+    | ``Images/Eval``                             |   Sampled examples of eval images     |
+    +---------------------------------------------+---------------------------------------+
+
+    Args:
+        interval (str, optional): Time string specifying how often to log train images. For example, ``interval='1ep'``
+            means images are logged once every epoch, while ``interval='100ba'`` means images are logged once every 100
+            batches. Eval images are logged once at the start of each eval. Default: ``"100ba"``.
+        mode (str, optional): How to log the image labels. Valid values are ``"input"`` (input only)
+            and "segmentation" which also saves segmentation masks. Default: ``"input"``.
+        num_images (int, optional): Number of images to log. Should be less than or equal to than the microbatch size.
+            If there are not enough images in the microbatch, all the images in the microbatch will be logged.
+            Default: ``8``.
+        input_key (str | int | Tuple[Callable, Callable] | Any, optional): A key that indexes to the input
+            from the batch. Can also be a pair of get and set functions, where the getter
+            is assumed to be first in the pair.  The default is 0, which corresponds to any sequence, where the first
+            element is the input. Default: ``0``.
+        target_key (str | int | Tuple[Callable, Callable] | Any, optional): A key that indexes to the target
+            from the batch. Can also be a pair of get and set functions, where the getter
+            is assumed to be first in the pair. The default is 1, which corresponds to any sequence, where the second
+            element is the target. Default: ``1``.
+    """
+
+    def __init__(self,
+                 interval: str = '100ba',
+                 mode: str = 'input',
+                 num_images: int = 8,
+                 input_key: Union[str, int, Tuple[Callable, Callable], Any] = 0,
+                 target_key: Union[str, int, Tuple[Callable, Callable], Any] = 1):
+        self.interval = interval
+        self.mode = mode
+        self.num_images = num_images
+        self.input_key = input_key
+        self.target_key = target_key
+
+        try:
+            import wandb
```

Comment:
Hmm, I'm not thrilled about having this callback only be compatible with wandb `wandb` here. That said, I realize that there isn't another way around this until we have the logger refactor.

I presume this PR will be ready before the logger refactor is in, and don't want that to hold this up. So, can you do the following?

1. Can you create a JIRA to track that ImageVisualizer will need to be refactored after the LoggerDesign refactor is implemented and assign it to @eracah?
2. Can you add a comment here -- e.g. `TODO(Evan): Generalize as part of the logger refactor`
3. Can you add a `.. note::` to the docstring that this callback only works with `wandb` logging for now
4. In the implementation, instead of calling `logger.data_batch()`, can you log the object directly to the wandb logger? E.g. `for dest in ensure_tuple(logger.destinations): if isinstance(dest, WandBLogger): dest.log_data(data)`? Mainly concerned about attempting to log a wandb.image to the file logger, and then having it either crash or blow up the logfile with a binary image.

Reply:
Yeah I don't love it either. I'd like to get this in soon, probably before the logger refactor because I think we might want to use it to demo our segmentation work. 

2-4 should be done in the code now. Will make the JIRA too.